content/blog/eng/lambda-days-2020.md

```diff
@@ -0,0 +1,91 @@
+categories: [English, Conference, Lambda-days, Krakow, Functional programming]
+title: "Lambda-days-2020"
+ita: "Lambda-days-2020"
+
+languageCode: "en-US"
+type: "post"
+---
+
+# Lambda Days 2021
+
+This year, Facile attended for the first time to [Lambda Days](https://www.lambdadays.org/lambdadays2020) Conference in the beautiful polish city of Krakow.
+It has been a two day full immersion, where the organizers have been able to gather speakers from all over the world
+showcasing the latest trends and academic research in the functional world, spanning from beginner approaches to functional languages 
+to super specific scientific applications.
+
+This conference totally worth attending, albeit very intense (sometimens mind blowing, actually!), the talks were high quality, and perfect organization.
+
+Here is a short list of talks that took our attention:
+
+- **Keynote: How to specify it! A guide to writing properties of pure functions**
+
+[Video](https://www.youtube.com/watch?v=G0NUOst-53U&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=3&t=0s)
+
+**John Hughes** one of Haskell creators presented his QuickCheck library
+In a funny manner, he showed how use his library to create automated test-cases and gave some tips on how to write tests that actually
+verify your assumptions and not giving false green check.
+He suggested first to write pure function as much as you can and creating properties by which test conditions can be automated.
+
+
+- **The power of Π**
+
+**Thorsten Altenkirch** from the University of Nottingham gave a talk about implementing dependent types
+Think for a second if it was possible to base return types in functions by the value at runtime. Normally you would think is not possible
+because we should be at runtime to know the actual input vale. The truth is that with some mathematical assumptions is possible to obtain
+type like non empty arrays that otherwise a compiler would not know
+
+[Video](https://www.youtube.com/watch?v=3zT5eVHpQwA&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=13&t=0s)
+
+- **Designing composable functional libraries, not just for data visualization**
+
+[Video](https://www.youtube.com/watch?v=G1Dp0NtQHeY&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=11&t=0s)
+
+**Tomas Petricek** from the Alan Turing Institute showed his tools library for data journalism
+He presented this library written in `f#`, and with a live demo built some graphs highlighting the endless possibilities given from the composability.
+
+
+- **Enhancing the type system with Refined Types**
+
+[Video](https://www.youtube.com/watch?v=Fx8WXcAZWuk&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=28&t=0s)
+
+**Juliano Alves**, Senior Software Engineer @TransferWise talked about refined types in Scala
+In this talks aimed to Scala beginners, he talked about how is achieve better return types.
+Through [Refined](https://github.com/fthomas/refined) library, which is a porting from the homonymus Haskell one, he showed how is possible
```

Comment:
Through [Refined](https://github.com/fthomas/refined) library, which is a porting from the homonymous Haskell one, he showed how it is possible to constrain types using predicates and how to create types like negative numbers and so on.

**Note:** does it sound ok to use "negative" instead of "below zero"?